firedrake/interpolation.py

```diff
@@ -286,28 +359,133 @@ def _interpolator(V, tensor, expr, subset, arguments, access):
         parloop_args.append(tensor(access, V.cell_node_map()))
     else:
         assert access == op2.WRITE  # Other access descriptors not done for Matrices.
-        parloop_args.append(tensor(op2.WRITE, (V.cell_node_map(),
-                                               arguments[0].function_space().cell_node_map())))
+        rows_map = V.cell_node_map()
+        columns_map = arguments[0].function_space().cell_node_map()
+        if trans_mesh:
+            # We have to map to the source mesh cells before mapping to nodes
+            columns_map = compose_map_and_cache(target_mesh_source_mesh_map,
+                                                columns_map)
```

Comment:
(Suggested) `target_mesh_source_mesh_map` -> ` target_mesh.cell_parent_cell_map`

Reply:
I am trying to account for a future case where the `target_mesh_source_mesh_map` map may not be the `target_mesh.cell_parent_cell_map`.

Reply:
I'll make this a single dispatch function I think

Reply:
Actually I think `target_mesh.cell_parent_cell_map` is fine